src/util/fields.py

```diff
@@ -1122,13 +1122,58 @@ def change_field_selection_values(cr, model, field, mapping, skip_inherit=()):
     table = table_of_model(cr, model)
 
     if column_exists(cr, table, field):
-        query = "UPDATE {table} SET {field}= %s::jsonb->>{field} WHERE {field} IN %s".format(table=table, field=field)
+        ctype = column_type(cr, table, field)
+        if ctype == "varchar":
+            query = "UPDATE {table} t SET {column} = %(json)s::jsonb->>t.{column} WHERE t.{column} = ANY(%(keys)s)"
+        elif ctype == "jsonb":
+            # company dependent selection field
+            query = """
+                WITH upd AS (
+                     SELECT t.id,
+                            jsonb_object_agg(v.key, COALESCE(%(json)s::jsonb->>v.value, v.value)) AS value
+                       FROM {table} t
+                       JOIN LATERAL jsonb_each_text(t.{column}) v
```

Comment:
> LATERAL can also precede a function-call FROM item, but in this case it is a noise word, because the function expression can refer to earlier FROM items in any case.

From PG [docs](https://www.postgresql.org/docs/current/sql-select.html#SQL-FROM)